tests/test_queries.py

```diff
@@ -622,3 +549,423 @@ def test_MapReduce(self):
         self.assertTrue(result["ok"], 1)
         self.assertTrue("counts" in result)
         self.assertTrue("results" in result)
+
+
+class TestInsertOne(_SingleCollectionTest):
+
+    @defer.inlineCallbacks
+    def test_Acknowledged(self):
```

Comment:
New `***_one/many` methods are using completely different protocol depending on whether current write-concern is acknowledged or not. If it is acknowledged, they are using new db commands which were introduced in MongoDB 2.6. When unacknowledged, they are using good old OP_INSERT/OP_UPDATE/... messages. So for each query methods we are testing ack'ed and unack'ed separately because message format and also additional flags handling (upsert, ordered, ...) are completely different.